Switched to using `RemoteChannel` as the outer `RemoteRef`.

rofinn opened the pull request:
Allows us to provide a `reset!` method that will recreate
the inner `RemoteChannel`. This can be useful when performing
error handling on `DeferredFuture`s to avoid deadlocks.

rofinn committed:
Switched to using `RemoteChannel` as the outer `RemoteRef`.

Allows us to provide a `reset!` method that will recreate
the inner `RemoteChannel`. This can be useful when performing
error handling on `DeferredFuture`s to avoid deadlocks.

iamed2 reviewed (changes requested):
Please update the README examples as well. Otherwise, looks good.

Commit:
Updated README.md

Updated the examples to reflect changing the outer `RemoteRef` to a `RemoteChannel`.
